install_config/persistent_storage/persistent_storage_flex_volume.adoc

```diff
@@ -0,0 +1,304 @@
+machines in the cluster, both masters and nodes as part of installation.
+
+[NOTE]
+====
+Flex Volume is an alpha feature and may change in a future release of {product-title}.
+====
+
+=== Installation
+
+A Flex Volume driver must be installed as an executable file
+`/usr/libexec/kubernetes/kubelet-plugins/volume/exec/<vendor>~<driver>/<driver>`.
+For example, OpenShift's driver for `foo` would be installed as an
+executable file named
+`/usr/libexec/kubernetes/kubelet-plugins/volume/exec/openshift.com~foo/foo`.
+
+This file must be present on all masters and nodes in the cluster.
+
+=== Usage
+
+Once the driver is installed, the storage it provides can be referenced from
+{product-title} using a `PersistentVolume`. Each `PersistentVolume` object
+in OpenShift represents one storage asset, typically a volume, in the
+storage backend.
+
+.Persistent Volume Object Definition Using FlexVolume
+====
+
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: pv0001 <1>
+spec:
+  capacity:
+    storage: 1Gi <2>
+  accessModes:
+    - ReadWriteOnce
+  flexVolume:
+    driver: openshift.com/foo <3>
+    fsType: "ext4" <4>
+    secretRef: foo-secret <5>
+    readOnly: true <6>
+    options: <7>
+      fooServer: 192.168.0.1:1234
+      fooVolumeName: bar
+----
+<1> The name of the volume. This will be how it is identified via
+xref:../../architecture/additional_concepts/storage.adoc#architecture-additional-concepts-storage[persistent volume
+claims] or from pods. This name may be different than the name of the volume
+on backend storage.
+<2> The amount of storage allocated to this volume.
+<3> Name of the driver. This field is mandatory.
+<4> Optional filesystem that is present on the volume.
+<5> Optional reference to a secret. Keys and value from this secret will be
+provided to the Flex Volume driver on invocation.
+<6> Optional read-only flag.
+<7> Additional options for the Flex Volume driver. This is a free-form
+dictionary of parameters that will be provided to the Flex Volume driver and
+typically includes name of the volume that this `PersistentVolume` represents,
+addresses of the storage server(s) and such.
+====
+
+=== Flex Volume drivers
+
+As noted above, a Flex Volume driver is an executable, typically a shell script,
+that resides in a well defined directory on all machines in the cluster, both
+masters and nodes. {product-title} calls it whenever it needs to attach, detach,
+mount or unmount a volume represented by a `PersistentVolume` with `flexVolume`
+as source.
+
+The first command line argument of the driver is always an operation name.
+Other parameters are specific to each operation. Most of the operations expect
+JSON string as a parameter. This parameter is literally a whole JSON string,
+not name of a file with the JSON! It contains all `flexVolume.options`, some
+options from `flexVolume` prefixed by `kubernetes.io/` and content of the
```

Comment:
Perhaps we can make this paragraph a list, to add some clarity. 

It contains:

    - All `flexVolume.options`
    - Some options from `flexVolume` prefixed by `kubernetes.io/
      ...